Firestore/core/src/firebase/firestore/model/document_set.h

```diff
@@ -44,6 +44,8 @@ class DocumentComparator : public util::FunctionComparator<Document> {
 
   static DocumentComparator ByKey();
 
+  using FunctionComparator<Document>::Compare;
```

Comment:
Is this used?  Did you maybe mean to change line 53 below to take advantage of this?

Reply:
This exists to make overloading work across class scopes. See http://www.stroustrup.com/bs_faq2.html#overloadderived.

Now that I've added it line 53 can use it, so I've simplified that.

I've also added a TODO to remove the using statement once the Objective-C overload is removed.

Reply:
Wow.  Super awesome!